pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -406,25 +422,47 @@ func (p *ConfigurationEditFlags) Apply(
 		}
 	}
 
-	if cmd.Flags().Changed("annotation") {
-		annotationsMap, err := util.MapFromArrayAllowingSingles(p.Annotations, "=")
+	if cmd.Flags().Changed("annotation") || cmd.Flags().Changed("annotation-service") || cmd.Flags().Changed("annotation-revision") {
```

Comment:
I can comment out the last condition in this if statement to read:

```go

if cmd.Flags().Changed("annotation") || cmd.Flags().Changed("annotation-service") { //|| cmd.Flags().Changed("annotation-revision") {	
...
}
```

Yet all tests are passing...

Reply:
Have refactored the tests